test/query/shorthand.test.ts

```diff
@@ -33,6 +35,124 @@ describe('query/shorthand', () => {
+      it('should correctly parse an encoding query given a channel and fieldDefShorthand', () => {
+        let specQ: SpecQuery = {mark: Mark.POINT, encodings: []} as SpecQuery;
+
+        shorthandParser.encoding(
+          specQ,
+          'x',
+          'bin(a,q,maxbins=20,scale={"type":"log"},sort={"field":"a","op":"mean","order":"descending"})'
+        );
+
+        assert.deepEqual(specQ.encodings[0], {
+          bin: {maxbins: 20},
+          channel: Channel.X,
+          field: 'a',
+          type: Type.QUANTITATIVE,
+          scale: {type: ScaleType.LOG},
+          sort: {field: 'a', op: AggregateOp.MEAN, order: SortOrder.DESCENDING}
+        });
+      });
+    });
+
+    describe('fn', () => {
+      it('should correctly parse an encoding query given a fieldDefShorthand with aggregation function', () => {
+        let encQ: EncodingQuery = {} as EncodingQuery;
+        shorthandParser.fn(encQ, 'sum(horsepower,q)');
+        assert.deepEqual(encQ, {aggregate: AggregateOp.SUM, field: 'horsepower', type: Type.QUANTITATIVE});
+      });
+
+      it('should correctly parse an encoding query given a fieldDefShorthand with count function', () => {
+        let encQ: EncodingQuery = {} as EncodingQuery;
+        shorthandParser.fn(encQ, 'count(model,n)');
+        assert.deepEqual(encQ,{aggregate: AggregateOp.COUNT, field: 'model', type: Type.NOMINAL});
+      });
+
+      it('should correctly parse an encoding query given a fieldDefShorthand with timeunit function', () => {
+        let encQ: EncodingQuery = {} as EncodingQuery;
+        shorthandParser.fn(encQ, 'hours(a,q)');
+        assert.deepEqual(encQ, {field: 'a', timeUnit: TimeUnit.HOURS, type: Type.QUANTITATIVE});
+      });
+
+      it('should correctly parse an encoding query given a fieldDefShorthand with bin function', () => {
+        let encQ: EncodingQuery = {} as EncodingQuery;
+        shorthandParser.fn(encQ, 'bin(a,q,maxbins=20,sort={"field":"a","op":"mean","order":"descending"})');
+        assert.deepEqual(encQ, {
+          bin: {maxbins: 20},
+          field: 'a',
+          type: Type.QUANTITATIVE,
+          sort: {field: 'a', op: AggregateOp.MEAN, order: SortOrder.DESCENDING}
+        });
+      });
+    });
+
+    describe('rawFieldDef', () => {
+      it('should correctly parse an encoding query from fieldDef parts', () => {
+        const encQ = shorthandParser.rawFieldDef({} as EncodingQuery,
+        splitWithTail('a,q,scale={"domain":["cats","dogs"],"exponent":3,"type":"pow"},axis={"orient":"top"},legend=false', ',', 2));
+        assert.deepEqual(encQ, {axis: {orient: AxisOrient.TOP}, field: 'a', legend: false, scale: {domain: ['cats', 'dogs'], exponent: 3, type: ScaleType.POW}, type: Type.QUANTITATIVE});
+      });
+    });
+
+    describe('calculate', () => {
+      it('should correctly parse calculate from shorthand', () => {
+        let specQ: SpecQuery = {} as SpecQuery;
+        let shorthandPart = 'calculate:{x2:datum.x*2},{x3:datum.x*3}';
```

Comment:
Why is it `calculate:{x2:datum.x*2},{x3:datum.x*3}` ????????????????????

It should be
```
calculate:{x2:datum.x*2, x3:datum.x*3}
```

If the old code is wrong, please file an issue.  Luckily we don't use calculate in our user study.